src/Datadog.Trace/Util/UriHelpers.cs

```diff
@@ -135,6 +172,84 @@ public static string GetCleanUriPath(string absolutePath)
 #endif
         }
 
+#if NETCOREAPP
+        internal static bool IsIdentifierSegment(ReadOnlySpan<char> segment)
+        {
+            if (segment.Length == 0)
+            {
+                return false;
+            }
+
+            var containsNumber = false;
+            foreach (var c in segment)
+            {
+                switch (c)
+                {
+                    case >= '0' and <= '9':
+                        containsNumber = true;
```

Comment:
There's a possible edge-case regression here: if something would have parsed as `Guid` before but had no numbers, we are no longer replacing it.
```
aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa
bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
```
Setting `containsNumber = true` on hyphens (but only if length is long enough?) takes care of the first case, but what about the second one?

My quick take is that is a string only has A-F and is longer than 16 chars, we should assume it's a hex string and replace it. On the other hand, my math (`6^32 / 16^32`) says only a very small fraction (2.3e-14) of all possible GUIDs contain letters only (ignoring other characters like hyphens). Maybe I'm overthinking this.

Reply:
Yeah, I considered that one, but as you say it's a very small fraction of all the GUIDs, but more importantly, _those aren't valid GUIDs_ 😉 Version 1, 2, 3, and 4 GUIDs all include the version as the first char in the middle group, i.e. a valid version 1 GUID would be `aaaaaaaa-aaaa-1aaa-aaaa-aaaaaaaaaaaa`, which would obviously be fine 🙂 . 

Another thing I considered was SHA hashes though, which seem like another good candidate. No those _could_ legitimately be all chars, but then those are looking at the v. small numbers you're talking about here too... hyhens obviously don't help with the hash thing though...

Not sure, I'm in two minds 🙂

Reply:
Yes, it's game of chance, I guess. How long must a string of `[A-F]` be until we're confident that it's meant to be hexadecimal? The current implementation is probably fine.